public/app/features/alerting/unified/RuleEditorCloudRules.test.tsx

```diff
@@ -0,0 +1,196 @@
+import userEvent, { PointerEventsCheckLevel } from '@testing-library/user-event';
+import React from 'react';
+import { Provider } from 'react-redux';
+import { Route, Router } from 'react-router-dom';
+import { clickSelectOption } from 'test/helpers/selectOptionInTest';
+import { byRole, byTestId } from 'testing-library-selector';
+
+import { selectors } from '@grafana/e2e-selectors';
+import { locationService, setDataSourceSrv } from '@grafana/runtime';
+import { contextSrv } from 'app/core/services/context_srv';
+import { configureStore } from 'app/store/configureStore';
+import { PromApplication } from 'app/types/unified-alerting-dto';
+
+import { searchFolders } from '../../manage-dashboards/state/actions';
+
+import RuleEditor from './RuleEditor';
+import { discoverFeatures } from './api/buildInfo';
+import { fetchRulerRules, fetchRulerRulesGroup, fetchRulerRulesNamespace, setRulerRuleGroup } from './api/ruler';
+import { ExpressionEditorProps } from './components/rule-editor/ExpressionEditor';
+import { disableRBAC, mockDataSource, MockDataSourceSrv } from './mocks';
+import { fetchRulerRulesIfNotFetchedYet } from './state/actions';
+import * as config from './utils/config';
+
+jest.mock('./components/rule-editor/ExpressionEditor', () => ({
+  // eslint-disable-next-line react/display-name
+  ExpressionEditor: ({ value, onChange }: ExpressionEditorProps) => (
+    <input value={value} data-testid="expr" onChange={(e) => onChange(e.target.value)} />
+  ),
+}));
+
+jest.mock('./api/buildInfo');
+jest.mock('./api/ruler');
+jest.mock('../../../../app/features/manage-dashboards/state/actions');
+
+// there's no angular scope in test and things go terribly wrong when trying to render the query editor row.
+// lets just skip it
+jest.mock('app/features/query/components/QueryEditorRow', () => ({
+  // eslint-disable-next-line react/display-name
+  QueryEditorRow: () => <p>hi</p>,
+}));
+
+jest.spyOn(config, 'getAllDataSources');
+
+const mocks = {
+  getAllDataSources: jest.mocked(config.getAllDataSources),
+  searchFolders: jest.mocked(searchFolders),
+  api: {
+    discoverFeatures: jest.mocked(discoverFeatures),
+    fetchRulerRulesGroup: jest.mocked(fetchRulerRulesGroup),
+    setRulerRuleGroup: jest.mocked(setRulerRuleGroup),
+    fetchRulerRulesNamespace: jest.mocked(fetchRulerRulesNamespace),
+    fetchRulerRules: jest.mocked(fetchRulerRules),
+    fetchRulerRulesIfNotFetchedYet: jest.mocked(fetchRulerRulesIfNotFetchedYet),
+  },
+};
+
+function renderRuleEditor(identifier?: string) {
+  const store = configureStore();
+
+  locationService.push(identifier ? `/alerting/${identifier}/edit` : `/alerting/new`);
+
+  return render(
+    <Provider store={store}>
+      <Router history={locationService.getHistory()}>
+        <Route path={['/alerting/new', '/alerting/:id/edit']} component={RuleEditor} />
+      </Router>
+    </Provider>
+  );
+}
+
+const ui = {
```

Comment:
Would be great if we extract this `ui` object related to AlertRuleForm so it could be reused in other tests.